api/turing/models/pyfunc_ensembler.go

```diff
@@ -0,0 +1,113 @@
+package models
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+type EnsemblerLike interface {
+	GetID() ID
+	GetProjectID() ID
+	GetType() EnsemblerType
+	GetName() string
+	GetCreatedAt() time.Time
+	GetUpdatedAt() time.Time
+
+	SetProjectID(id ID)
+	Patch(other EnsemblerLike) error
+}
+
+func EnsemblerTable(ensembler EnsemblerLike) func(tx *gorm.DB) *gorm.DB {
+	return func(tx *gorm.DB) *gorm.DB {
+		switch ensembler.GetType() {
+		case EnsemblerTypePyFunc:
+			return tx.Table("pyfunc_ensemblers")
+		default:
```

Comment:
Is this expected? Since there is no ensembler config related info in the `ensemblers` table, I imagine that table is just used to store the common fields. Should the default case raise an error?

Reply:
From the service perspective (and generally from the inherited db tables perspective) – this is expected. From the application perspective this doesn't really make sense (because generic ensemblers aren't useable), and that's why we have [validation](https://github.com/romanwozniak/turing/blob/a05dc201c839a036e3c2f05c24939fe22d46d7d7/api/turing/api/ensemblers_api.go#L174) that checks the type of ensembler and rejects ensembler with unsupported types